dlq_processor/sumo-dlq-function-utils/lib/SumoLogsClient.js

```diff
@@ -0,0 +1,84 @@
+var https = require('https');
+var url = require('url');
+
+function SumoLogsClient(config) {
+    this.options = config.options || {};
+    if (config.SumoURL) {
+        var urlObj = url.parse(config.SumoURL);
+        this.options.hostname = urlObj.hostname;
+        this.options.path = urlObj.pathname;
+        this.options.protocol = urlObj.protocol;
+    }
+    this.options.method = 'POST';
+}
+
+function generateHeaders(config, message, awslogsData) {
+    var sourceCategory = config.sourceCategoryOverride || '';
+    var sourceName = config.sourceNameOverride || ((awslogsData) ? awslogsData.logStream : '');
+    var sourceHost = config.sourceHostOverride || ((awslogsData) ? awslogsData.logGroup : '');
+
+    var headerObj = {
+        'X-Sumo-Name':sourceName, 'X-Sumo-Category':sourceCategory,
+        'X-Sumo-Host':sourceHost, 'X-Sumo-Client': config.SUMO_CLIENT_HEADER
+    };
+
+    var metadataMap = { 'source': 'X-Sumo-Name', 'category': 'X-Sumo-Category', 'host': 'X-Sumo-Host'};
+    if (message.hasOwnProperty('_sumo_metadata')) {
+        var metadataOverride = message._sumo_metadata;
+        Object.getOwnPropertyNames(metadataOverride).forEach( function(property) {
+            if (metadataMap[property]) { //format for metadatamap keys soureName(in azure) or name
+                var targetProperty = metadataMap[property];
+                headerObj[targetProperty] = metadataOverride[property];
+            }
+        });
+        delete message._sumo_metadata;
+    }
+    return headerObj;
+}
+
+SumoLogsClient.prototype.postToSumo = function (messages, errorHandler, beforeRequest) {
+    var messagesTotal = Object.keys(messages).length;
+    var messagesSent = 0;
+    var messageErrors = [];
+
+    var finalizeContext = function () {
```

Comment:
This used to be called finalizeContext because it's called last before existing. Should change to updateContext.